source/Calamari.Shared/Integration/Packages/Download/PackageDownloaderStrategy.cs

```diff
@@ -40,18 +43,18 @@ public PackageDownloaderStrategy(IScriptEngine engine, ICalamariFileSystem fileS
             switch (feedType)
             {
                 case FeedType.Maven:
-                    downloader = new MavenPackageDownloader();
+                    downloader = new MavenPackageDownloader(fileSystem, freeSpaceChecker);
```

Comment:
Should the free space check & loading from cache be done by the strategy? that logic seems to be repeated (and missing) in some downloaders

Reply:
I did consider that, but Helm packages are not significant in size. Not sure about the Docker ones. Do they work on a remote target @MJRichardson  so local disk space isn't relevant?

Reply:
If it's not applicable for some, then we could declare it on `IPackageDownloader` and handle it in the strategy